pymmcore_plus/client/_client.py

```diff
@@ -111,23 +95,22 @@ def new_server_process(
 
 
 def ensure_server_running(
-    host, port, timeout=5, verbose=False, cleanup_new=True, cleanup_existing=False
+    host, port, timeout=5, verbose=False, cleanup_new=True
 ) -> Optional[subprocess.Popen]:
     """Ensure that a server daemon is running, or start one."""
     uri = f"PYRO:{core.DAEMON_NAME}@{host}:{port}"
     remote_daemon = api.Proxy(uri)
     try:
         remote_daemon.ping()
         logger.debug("Found existing server:\n{}", remote_daemon.info())
-        if cleanup_existing:
-            proc = _get_remote_pid(host, port)
-            if proc is not None:
-                atexit.register(proc.kill)
     except errors.CommunicationError:
         logger.debug("No server found, creating new mmcore server")
         proc = new_server_process(host, port, verbose=verbose)
         if cleanup_new:
             atexit.register(proc.kill)
+            atexit.register(
```

Comment:
does order matter here?  should we register `unload` before proc.kill?

Reply:
Definitely matters - it's important that  `unload` is registered after because `atexit` calls functions in reverse order of registration.

Reply:
thanks!